src/integration-tests/tests/serialization.rs

```diff
@@ -41,6 +41,54 @@ mod serialization {
         Ok(())
     }
 
+    #[test]
+    fn serde_with_float() -> Result<()> {
```

Comment:
optional: these are only `ser`, consider validating the `de`:

```rs
let roundtrip = serde_json::from_value::<T>(got)?;
assert_eq!(input, roundtrip); // or whatever we call the original message
```

Reply:
done!